jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/EnableEcsExecuteCommand.kt

```diff
@@ -0,0 +1,44 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.ecs.exec
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.project.Project
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import software.amazon.awssdk.services.ecs.model.Service
+import software.aws.toolkits.jetbrains.core.explorer.actions.SingleResourceNodeAction
+import software.aws.toolkits.jetbrains.services.ecs.EcsServiceNode
+import software.aws.toolkits.jetbrains.settings.EcsExecCommandSettings
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.jetbrains.utils.notifyInfo
+import software.aws.toolkits.resources.message
+
+class EnableEcsExecuteCommand :
+    SingleResourceNodeAction<EcsServiceNode>(message("ecs.execute_command_enable"), null),
+    CoroutineScope by ApplicationThreadPoolScope("EnableExecuteCommand") {
+    private val settings = EcsExecCommandSettings.getInstance()
+    override fun actionPerformed(selected: EcsServiceNode, e: AnActionEvent) {
+        if (!settings.showExecuteCommandWarning || ExecuteCommandWarning(selected.nodeProject, enable = true).showAndGet()) {
+            launch {
+                enableExecuteCommand(selected.nodeProject, selected.value)
+            }
+        }
+    }
+
+    override fun update(selected: EcsServiceNode, e: AnActionEvent) {
+        e.presentation.isVisible = !EcsExecUtils(selected.nodeProject).executeCommandFlagStatus(selected.value)
```

Comment:
This looks like on a cache miss it will trigger a network call on the UI and block it, why do we need to if the EcsServiceNode already should have the data on it from the original describe?

Reply:
In case the user changes the status of the enableExecuteCommand flag externally, the enable/disable option will be correctly displayed based on the most recent status of the flag.

Reply:
We accept that risk today, since network calls blocking the UI is worse than it being out of date, especially if validated when the action is ran.

When the enable/disable tasks are ran we typically refresh the subpart of the explorer, see usages of `refreshAwsTree`

Reply:
The current push refreshes the service after updating it. The flag status is now updated with the service node describe call.